src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (number) => {
+    if (number || number === 0) {
+      sum += number;
+
+      return adder;
+    } else {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
```

Comment:
```suggestion
    }
    
    const result = sum;

    sum = 0;

    return result;
   
```
else is redundant here